hashintel/hash
# Support lists/arrays on entity editor

## 🌟 What is the purpose of this PR?

This PR adds list/array editing support on Entity editor. There will be follow-up PRs to add multi-type support and UX improvements to this.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203150072758509/1203199644802788/f) _(internal)_
- [Design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-App?node-id=6295%3A253013&t=AcMTJsXFjdQWI8tA-4) _(internal)_

## 🔍 What does this change?

- `PropertyRow`
  - add `isArray`
  - rename `dataTypes` to `expectedTypes`
  - fix `expectedTypes`, now is shows types expected types in an array instead of showing `Array` as type
- Add `bpList` icon to `custom-grid-icons`

## 🐾 Next steps

- Improve mobile UX (e.g. test drag-drop on mobile)
- Support multiple types on array-editor
- Lock page scroll when the editor is open

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Apply https://github.com/hashintel/hash/commit/19c2f7fd0b4426e69ac744848fcf63b5b7089b2f this commit on your local to get a button on entity editor, which creates a complex entity type for you to test this 
1.  Confirm that we can edit lists on entity editor

## 📹 Demo

⚠️ Ignore the gray page opens at 0:42, that's the break-reminder app on my computer 😄 

https://user-images.githubusercontent.com/39553853/203017102-84971856-1af2-4a01-9b0c-1440e4119062.mov
